### PR TITLE
doc: add more installation methods

### PR DESCRIPTION
**Pull request by kemingy**
Signed-off-by: Keming <kemingyang@tensorchord.ai>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,10 +111,17 @@ For example, the PyPI cache is shared across builds and thus the package will be
...
+`envd` can be installed with `pip`, or you can download the binary [release](https://github.com/tensorchord/envd/releases) directly. After the installation, please run `envd bootstrap` to bootstrap.
 
 ```bash
-pip3 install --pre --upgrade envd
+# install with local pip
+pip install --pre --upgrade envd
+# or install with pipx
+pipx install envd --pip-args="--pre"
+# or download the latest release artifact with GitHub CLI
+bash -c "gh release download -R tensorchord/envd -p $(echo envd_\*_$(uname)_$(uname -p))"
```
Comment (gaocegege) [inline README.md:122]: What will happen if there is no gh installed?

Comment (kemingy) [reply README.md:122]: Why will you run this command if you don't have gh installed?

Comment (gaocegege) [reply README.md:122]: because I do not know what the command means from a ds's perspective.

Comment (kemingy) [reply README.md:122]: See the comments, `with GitHub CLI`.

I don't think you will run `pipx` if you don't have `pipx`. The same to `gh`.

**Comment by frostming** [comment]
It surprises me that there is no 'installation' section in the docs, or is there one that I missed?

IMHO we should list recommended installation methods together with alternatives in the docs, and keep the installation part in README simple.

**Comment by kemingy** [comment]
> It surprises me that there is no 'installation' section in the docs, or is there one that I missed?
> 
> IMHO we should list recommended installation methods together with alternatives in the docs, and keep the installation part in README simple.

The current [installation section](https://github.com/tensorchord/envd#install-and-bootstrap-envd) doesn't have enough instructions for beginners. Users may not know "how to add it to the PATH" or "why I'm using the old one even after `pip install -U --pre envd`".

I agree we should improve the installation section.

**Comment by frostming** [comment]
@kemingy Yeah, we can add an installation section to [Building Your First Environment page](https://envd.tensorchord.ai/guide/build-envd.html), or in its own page. We can then make it more verbose and clear there, and leave a link referring to the docs in README.

**Comment by gaocegege** [comment]
BTW, should we remove --pre from the script? Since we will follow the semver to publish new releases.

**Comment by kemingy** [comment]
> BTW, should we remove --pre from the script? Since we will follow the semver to publish new releases.

Agree. cc @frostming you can delete `--pre` when you're trying to work on https://github.com/tensorchord/envd-docs/issues/204

**Pull request merged**